util/src/journaldb.rs

```diff
@@ -24,6 +24,34 @@ use kvdb::{Database, DBTransaction, DatabaseConfig};
 #[cfg(test)]
 use std::env;
 
+#[derive(Clone, PartialEq, Eq)]
+struct RefInfo {
```

Comment:
Idea for future optimization: compact this struct. Too bad rust does not support bit fields. In my original implementation I've used `u16` for counter and `#[repr(packed)]` for the structure to reduce memory usage. It is also possible to use a single bit of `queue_refs` for the flag